src/ring_theory/graded_algebra/homogeneous_ideal.lean

```diff
@@ -471,3 +471,84 @@ lemma ideal.homogeneous_hull_eq_Inf (I : ideal A) :
 eq.symm $ is_glb.Inf_eq $ (ideal.homogeneous_hull.gc 𝒜).is_least_l.is_glb
 
 end galois_connection
+
+section irrelevant_ideal
+
+open graded_algebra finset set_like.graded_monoid
+
+variables [comm_semiring R] [semiring A]
+variables [algebra R A] [decidable_eq ι]
+variables [canonically_ordered_add_monoid ι] [covariant_class ι ι has_add.add has_lt.lt]
+variables (𝒜 : ι → submodule R A) [graded_algebra 𝒜]
+
+/--
+For a graded ring `⨁ᵢ 𝒜ᵢ` graded by a `canonically_ordered_add_monoid ι`, the irrelevant ideal
+refers to `⨁_{i≥0} 𝒜ᵢ`, or equivalently `{a | the bot-th projection of a = 0}`.
+
+This definitoin is used in Proj construction where `ι` is always `ℕ` so the irrelevant ideal is
+simply elements with `0` as 0-th coordinate. But the notion of irrelevant ideal makes sense in a
+more general setting by defining it as the ideal of elements with `0` as i-th coordinate for all
+`i ≤ 0`.
+-/
+def homogeneous_ideal.irrelevant : homogeneous_ideal 𝒜 :=
+let f : A →+* A :=
+{ to_fun := λ a, proj 𝒜 0 a,
+  map_one' := by { rw [proj_apply, decompose_of_mem_same], exact set_like.graded_monoid.one_mem },
+  map_mul' := λ x y, begin
+    haveI : Π (i : ι) (x : (𝒜 i)), decidable (x ≠ 0) := λ _ _, classical.dec_pred _ _,
+    apply @submodule.supr_induction _ _ _ _ _ _ 𝒜
+        (λ x, proj 𝒜 0 (x * y) = proj 𝒜 0 x * proj 𝒜 0 y),
```

Comment:
You can avoid the `@`s here any below if you start with
```
    have hx : x ∈ ⨆ i, 𝒜 i := ((is_internal 𝒜).supr_eq_top.ge : _) submodule.mem_top,
    have hy : y ∈ ⨆ i, 𝒜 i := ((is_internal 𝒜).supr_eq_top.ge : _) submodule.mem_top,
```
which also seems more readable anyway.

Reply:
I used `@` mainly to spell out the induction premise `(λ x, proj 𝒜 0 (x * y) = proj 𝒜 0 x * proj 𝒜 0 y)`; otherwise Lean will find a weird one.

Reply:
It seems to find the right one if you pass `hx` to it explicitly - or at least, your proof seems to work unchanged.

Reply:
Thanks!